### ISSUE TITLE
Pyright 1.1.385 fails to infer the type where it originally could

### ISSUE THREAD

**Issue by AlexanderPodorov** [microsoft/pyright#9255]
**Describe the bug**
See MRE

```python
from typing import reveal_type

data = {
    "a": 1,
    "b": 1.0,
    "c": "d",
}

reveal_type(data)  # Type of "data" is "dict[str, Unknown]"
# However it should be dict[str, int | float | str]

```

**VS Code extension or command-line**
Pylance v2024.10.101 (pre-release), Pyright 1.1.385.

**Comment by erictraut** [comment]
Pyright's behavior hasn't changed here. Refer to [this documentation](https://microsoft.github.io/pyright/#/type-inference?id=dictionary-expressions) for details on how pyright's inference works for dictionary expressions. I'm guessing that you previously had `strictDictionaryInference` enabled either explicitly or implicitly via the "strict" type checking mode.

**Comment by AlexanderPodorov** [comment]
@erictraut , thanks a lot for pointing me to the right direction! I was confused because previously I did not need to set this settings to `true` explicitly, this probably was set to `true` by default, but as I see it's set to `false` by default. Probably it has something to do with Pylance or Python extension.
